tests/async_integration_tests.py

```diff
@@ -482,231 +482,114 @@ async def test_add_and_update_file_to_archive_item(local_grand_challenge):
-
         # create new archive item
-        _ = await c.upload_cases(
+        archive_item_pks = await c.add_cases_to_archive(
             archive="archive",
-            files=[Path(__file__).parent / "testdata" / "image10x10x101.mha"],
+            archive_items=[
+                {"generic-medical-image": TESTDATA / "image10x10x101.mha"},
+            ],
         )
 
-        # retrieve existing archive item pk
-        items = await get_archive_items(c, archive.pk, len(old_items_list))
-
-        old_items_pks = {item.pk for item in old_items_list}
-
-        # get the new item
-        target_archive_item = None
-        for item in items:
-            if item.pk not in old_items_pks:
-                target_archive_item = item
-                break
+        @async_recurse_call
+        async def get_target_archive_item():
+            # Wait for the image to be added (async task)
+            archive_item = await c.archive_items.detail(pk=archive_item_pks[0])
+            if len(archive_item.values) != 1:
+                # item has not been added
+                raise ValueError
+            return archive_item
 
-        # Sanity
-        assert "predictions-csv-file" not in [
-            item.interface.slug for item in target_archive_item.values
-        ]
-
-        with pytest.raises(ValueError) as e:
-            _ = await c.update_archive_item(
-                archive_item_pk=target_archive_item.pk,
-                values={
-                    "predictions-csv-file": [
-                        Path(__file__).parent / "testdata" / f
-                        for f in ["test.csv", "test.csv"]
-                    ]
-                },
-            )
-        assert (
-            "You can only upload one single file to a predictions-csv-file interface"
-            in str(e)
-        )
+        target_archive_item = await get_target_archive_item()
 
+        # Update it again, with a CSV this time
         _ = await c.update_archive_item(
             archive_item_pk=target_archive_item.pk,
             values={
-                "predictions-csv-file": [
-                    Path(__file__).parent / "testdata" / "test.csv"
-                ]
+                "predictions-csv-file": [TESTDATA / "test.csv"],
             },
         )
 
         @async_recurse_call
         async def get_updated_archive_item():
             archive_item = await c.archive_items.detail(target_archive_item.pk)
-            if len(archive_item.values) != len(target_archive_item.values) + 1:
+            if len(archive_item.values) != 2:
                 # item has not been added
                 raise ValueError
             return archive_item
 
-        updated_item = await get_updated_archive_item()
+        item_updated = await get_updated_archive_item()
 
-        updated_civs_by_slug = {
-            item.interface.slug: item for item in updated_item.values
-        }
+        csv_socket_value = item_updated.values[0]
```

Comment:
Are items guaranteed to be sorted this way?

Reply:
I think so ... several other tests rely on. However: better to not assume or rely on this.